### PR TITLE
Fix missing cache skip in proj 6 code path

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: src/core/qgscoordinatetransform.cpp [hunk 1]
```diff
@@ -821,7 +821,10 @@ bool QgsCoordinateTransform::setFromCache( const QgsCoordinateReferenceSystem &s
   if ( sourceKey.isEmpty() || destKey.isEmpty() )
     return false;
 
-  sCacheLock.lockForRead();
+  QgsReadWriteLocker locker( sCacheLock, QgsReadWriteLocker::Read );
```
Comment (m-kuhn) [inline src/core/qgscoordinatetransform.cpp]: Could this be moved to after the condition to avoid the locking/unlocking in case caching is disabled?

Comment (nyalldawson) [reply src/core/qgscoordinatetransform.cpp]: I'm trying to be super-safe here and also protect the disable cache static boolean value.

Comment (m-kuhn) [reply src/core/qgscoordinatetransform.cpp]: Hmm... ok.
I wouldn't expect issues (if there's a concurrent write at the same time changing the value the result will be a matter of timing with or without locking). But yes it still feels slightly cleaner.

Comment (nyalldawson) [reply src/core/qgscoordinatetransform.cpp]: I'm still inclined to be overly safe here, especially given that the cache is only ever disabled on shutting down qgis, and we would ideally expect this situation to never arise (accessing cache during the shutdown process).

Comment (m-kuhn) [reply src/core/qgscoordinatetransform.cpp]: All good :)

Comment (nyalldawson) [reply src/core/qgscoordinatetransform.cpp]: Let's just hope this finally puts this issue to bed...

Comment (m-kuhn) [reply src/core/qgscoordinatetransform.cpp]: fingers crossed!

Comment (nirvn) [reply src/core/qgscoordinatetransform.cpp]: Going to a nearby pagoda to light up so incense! ;)

**Pull request merged**